### ISSUE TITLE
Semi-auto Ninjutsu Mudra for Ninja (No Bunny Feature / No Bunny Alternate Feature)

### ISSUE THREAD

**Issue by NikkoToska** [Nik-Potokar/XIVSlothCombo#266]
I think it should be possible to semi-automate casting ninjutsu so that you can't fail, if it's possible to track the Ninjutsu action state.  So if you can detect that after one mudra is pressed that Ninjutsu has become Fuma, then this should work:
```
Initial State -> 1st Press -> 2nd Press -> 3rd Press
----------------------------------------------------
Ten           -> Jin       -> Chi       -> Ninjutsu
Chi           -> Ten       -> Jin       -> Ninjutsu
Jin           -> Chi       -> Ten       -> Ninjutsu
```
This way it's easier to activate ninjutsu and should be impossible to mess up.  You might choose the wrong one, but if you just tap one button, the ninjutsu will go off.

So, for example:
For Fuma, I can press any mudra once, and I can press the Ninjutsu button to activate.
For Hyoton, I tap the Ten button twice, then press Ninjutsu to activate.
For Doton, I tap the Ten button three times, at which point all mudra become Ninjutsu, so one more tap will activate.

I think it would be good if there was an alternate version as well in the case that some people find it easier or more convenient to press a different button to start.  So the alternate would be:
```
Initial State -> 1st Press -> 2nd Press -> 3rd Press
----------------------------------------------------
Ten           -> Chi       -> Jin       -> Ninjutsu
Chi           -> Jin       -> Ten       -> Ninjutsu
Jin           -> Ten       -> Chi       -> Ninjutsu
```
It's only semi-auto since you still have to activate the tier two ninjutsu with a different button, but as long as you only press the same button, the ninjutsu won't fail.

**Comment by nativepapaya** [comment]
This is pretty neat!

**Comment by Iaotle** [comment]
Not sure if this is possible with the current toolset.

**Comment by Iaotle** [comment]
@augporto u think this is possible?

**Comment by augporto** [comment]
I think that may be possible yes, since ninjutsus are now gcds (I think), ill take a look at it as soon as I can.

**Comment by augporto** [comment]
I made a PR that should implement the 1st example @NikkoToska proposed, if demand appear for the second one ill make it.

It need some feed back for high lv ninjutsu , the ones at lv76, since my ninja is not that high lv , but it should account for it.

**Comment by NikkoToska** [comment]
I tested it with the high level ninjutsu and it works perfectly, even with kassatsu and ten-chi-jin.  Even though it was my idea, it takes a little mental gymnastics to get used to the buttons.

One possible issue could be after the first button press.  All the buttons rotate, but the other buttons don't become ninjutsu like they do after the second press. For example, if I press the button in the TEN position twice, the buttons in the buttons formerly in the CHI and JIN become ninjutsu of the appropriate type.  But if I only press TEN once, it becomes CHI, but I could still accidentally press TEN again in the old CHI position.  Is it possible to know what button was pressed first, or is it only after the second press when the ninjutsu reveals what must have been pressed previously?  The chat log does indicate the mudra used if that can be parsed.

**Comment by augporto** [comment]
> I tested it with the high level ninjutsu and it works perfectly, even with kassatsu and ten-chi-jin. Even though it was my idea, it takes a little mental gymnastics to get used to the buttons.
> 
> One possible issue could be after the first button press. All the buttons rotate, but the other buttons don't become ninjutsu like they do after the second press. For example, if I press the button in the TEN position twice, the buttons in the buttons formerly in the CHI and JIN become ninjutsu of the appropriate type. But if I only press TEN once, it becomes CHI, but I could still accidentally press TEN again in the old CHI position. Is it possible to know what button was pressed first, or is it only after the second press when the ninjutsu reveals what must have been pressed previously?

Good to know it works on high lv too :)

I just took a look and yea you are right , I thought it was not changing the others on the 1st press, but I guess I was wrong, will take a look at it.

**Comment by NikkoToska** [comment]
After thinking about it and trying it out, I thought of another option.  I think there would be a lot of people who won't like that the mudra change location and muscle memory would be hard to overcome.  How about after each mudra is pressed, only that mudra is changed to ninjutsu?  That way you can still follow the pattern that you're used to using to cast ninjutsu, but you won't be able to double press any mudra.  There can still be a "simple mudra" way, but I think it would be more approachable to more people not to move them around, but you still can't mess up and you can free up one button.  What do you think?

**Comment by augporto** [comment]
@NikkoToska I remember now why the 1st cast behave like that.

Its because the only way I found to track the state is with the resulting ninjutsu, and since all of them in the 1st press return fuma, I cant differentiate which mudra was the one that casted it. 

I've digged a little on the interns of toolset but did not find anything that could help right now with it.

Unless someone with more experience with the toolset can give an idea I don't think I can do much on the way of the 1st cast, except be careful with fat fingering the other mudra, since you only press 1 mudra to make a ninjutsu I don't think that would be too hard, but still would need some muscle memory to get used to.

I'll still look at it when I can , but can not guarantee right now that it can be done like that :(
